### PR TITLE
[SPARK-45345][BUILD] Clean Scala version change leftovers in  `release-build.sh`

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to clean Scala version change leftovers in  `release-build.sh`ean 

### Why are the changes needed?

For the proper releases.

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

Manually tested the changes.

### Was this patch authored or co-authored using generative AI tooling?

No,

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @LuciferYang @dongjoon-hyun

**Comment by LuciferYang** [comment]
https://github.com/apache/spark/blob/cc4ecb5104e37d5e530d44b41fc1d8f8116e37d8/dev/create-release/release-build.sh#L332-L350

Doesn't the logic above need to be fixed?

File: dev/create-release/release-build.sh [hunk 1]
```diff
@@ -186,6 +186,9 @@ if [[ $SPARK_VERSION < "3.2" ]]; then
 fi
 
 PUBLISH_SCALA_2_12=1
+if [[ $SPARK_VERSION < "4.0" ]]; then
+  PUBLISH_SCALA_2_12=0
+fi
```
Comment (beliefer) [inline dev/create-release/release-build.sh:191]: It seems PUBLISH_SCALA_2_12 and PUBLISH_SCALA_2_13 will be 0, if `SPARK_VERSION < "3.2"`.

**Review by beliefer — APPROVED** [review]
LGTM except one comment.

**Comment by LuciferYang** [comment]
https://github.com/apache/spark/blob/cc4ecb5104e37d5e530d44b41fc1d8f8116e37d8/dev/create-release/release-build.sh#L199-L203

This logic is unrelated to Scala, but it seems necessary to add a Java version check for 4.0+, as it will be released by Java 17. Maybe in a separate pr?

**Comment by HyukjinKwon** [comment]
Manually tested.

Merged to master.

**Pull request closed without merging**